pkg/middleware/region_proxy.go

```diff
@@ -6,33 +6,64 @@ import (
 	"net/http"
 	"net/http/httputil"
 	"net/url"
+	"strings"
+	"time"
 
+	"github.com/hashicorp/golang-lru/v2/expirable"
 	sdk "github.com/openshift-online/ocm-sdk-go"
 	"github.com/openshift-online/ocm-sdk-go/logging"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const (
+	// region ids
+	APSoutEast1 = "ap-southeast-1"
+	USEast1     = "us-east-1"
+	Global      = "global"
+
+	defaultCacheSize       = 1000
+	defaultCacheExpireTime = time.Hour * 24
```

Comment:
clusters will *never* migrate to a different region.  why would we ever need to expire the cache?

Reply:
it's due to the memory consumption since currently it's a big concern for CS runtime, per Matt's above comment: https://github.com/openshift-online/ocm-service-common/pull/3#discussion_r2069697886

Reply:
current num subs in prod: 489297
current numb clusters in prod: 464030

to Mark's point we dont actually care about expiring cache entries. The reason i suggested the LRU cache is 100% because we can explicitly set a max cache size (number of entries) and when that limit is reach it will evict the least recently used cache entry. It is a trivial way of eliminating the possibility of cache memory bloat over very long time spans (run time of a pod and the life time of OCM service development...total number of clusters only goes up...)

It would be perfectly fine to never expire cache entries on and only have a max size declared, but its a reasonable precaution to have some upper limit.